lib/ansible/plugins/filter/core.py

```diff
@@ -206,7 +206,12 @@ def regex_escape(string):
 
 def from_yaml(data):
     if isinstance(data, string_types):
-        return yaml.safe_load(data)
+        try:
+            return yaml.safe_load(data)
+        except yaml.composer.ComposerError as e:
+            if e.context == 'expected a single document in the stream':
```

Comment:
I do hate being dependent on strings in other projects, but there are a lot of composer errors that can be raised. We could retry every time and I think the proper error would be re-raised, but I'm not confident enough to bet on that.

Reply:
Also if we support alternative yaml libraries I'd need to ensure that we can handle that